java-util/src/main/java/io/druid/java/util/common/concurrent/SameThreadExecutorService.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.java.util.common.concurrent;
+
+import java.util.List;
+import java.util.concurrent.AbstractExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A simple class that implements the ExecutorService interface, but runs the code on a call to submit
+ */
+public class SameThreadExecutorService extends AbstractExecutorService
```

Comment:
Please make this class a singleton

Reply:
I made it singleton in `Execs`

Reply:
Execs doesn't prohibit instantiation of new `SameThreadExecutorService` objects.

Reply:
Yes that is true. Why force singleton?

Reply:
To prohibit accidential instantiation.

Referencing Execs in the javadoc comment would be useful too.

Reply:
Made the class constructor package private and added some more comments.